aptos-move/framework/table-natives/src/lib.rs

```diff
@@ -534,14 +527,36 @@ fn get_table_handle(table: &StructRef) -> PartialVMResult<TableHandle> {
     Ok(TableHandle(handle))
 }
 
-fn serialize(layout: &MoveTypeLayout, val: &Value) -> PartialVMResult<Vec<u8>> {
-    val.simple_serialize(layout)
-        .ok_or_else(|| partial_extension_error("cannot serialize table key or value"))
+fn serialize_key(layout: &MoveTypeLayout, key: &Value) -> PartialVMResult<Vec<u8>> {
+    key.simple_serialize(layout)
+        .ok_or_else(|| partial_extension_error("cannot serialize table key"))
 }
 
-fn deserialize(layout: &MoveTypeLayout, bytes: &[u8]) -> PartialVMResult<Value> {
-    Value::simple_deserialize(bytes, layout)
-        .ok_or_else(|| partial_extension_error("cannot deserialize table key or value"))
+fn serialize_value(
+    layout_info: &LayoutInfo,
+    val: &Value,
+) -> PartialVMResult<(Bytes, Option<Arc<MoveTypeLayout>>)> {
+    let serialization_result = if layout_info.has_identifier_mappings {
+        // Value contains delayed fields, so we should be able to serialize it.
+        serialize_and_allow_delayed_values(val, layout_info.layout.as_ref())
+            .map(|bytes| (bytes.into(), Some(layout_info.layout.clone())))
+    } else {
+        // No delayed fields, make sure serialization fails if there are any
```

Comment:
I might have asked this before, does simple serialize already fail if there are native values?